ql/src/java/org/apache/hadoop/hive/ql/parse/MergeSemanticAnalyzer.java

```diff
@@ -62,11 +62,10 @@ protected ASTNode getTargetTableNode(ASTNode tree) {
 
   @Override
   public void analyze(ASTNode tree, Table targetTable, ASTNode tableNameNode) throws SemanticException {
-    if (tree.getToken().getType() != HiveParser.TOK_MERGE) {
-      throw new RuntimeException("Asked to parse token " + tree.getName() + " in " +
-          "MergeSemanticAnalyzer");
+    boolean nonNativeAcid = AcidUtils.isNonNativeAcidTable(targetTable);
+    if (nonNativeAcid) {
```

Comment:
Could we create an automatic way to turn on the appropriate configs when the query is an Iceberg query? Maybe something like `HiveIcebergStorageHandler.fallbackToNonVectorizedModeBasedOnProperties`?

Reply:
It is already implemented and used independently from this patch. `HiveIcebergStorageHandler.fallbackToNonVectorizedModeBasedOnProperties` is called when FSOperator "augments the plan"
https://github.com/apache/hive/blob/4146d220bfc0a29d117676844a4c7021fcc6a30d/ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java#L1612